src/data_writer.py

```diff
@@ -52,20 +53,20 @@ def invalidColors(file):
 
     except FileNotFoundError:
         return invalidList
-def colorPercent(file,color):
-    try:
-        colors = numOfEachColor(file)
-        totalCalls = sum(colors.values())
-        if totalCalls == 0:
-            totalCalls = 1;
-        if(colors.get(color) != None):
-            percent = colors.get(color) / totalCalls * 100
-        else:
-            percent = 1/totalCalls*100
-        return percent
-    except FileNotFoundError:
-        return 0
-def firstEntryDate(file):
+
+
+def color_percent(color):
+    r = redis.Redis(host='localhost', port=6379, db=0)
+    color = color.lower()
+
+    color_total = (float(r.hget('color_totals', color).decode('utf-8'))) + 1 #The addition is to make sure the output for the first random is not 0%
```

Comment:
I don't understand the +1 here.  Can you explain the issue with random a bit more?

It seems like if there is an issue with random, it should be special-cased (if color == 'random: ...) rather than doing +1 for all cases.